services/supervisor/src/transactions.rs

```diff
@@ -141,11 +141,21 @@ impl SupervisorInterface for Supervisor {
             return Err(Error::ActualFromIsPast.into());
         }
 
-        // Verifies that there are no pending config changes.
-        if schema.pending_proposal().exists() {
+        // Verifies that there are no pending config changes assigned to the same height.
+        if schema.pending_proposals().contains(&propose.actual_from) {
             return Err(Error::ConfigProposeExists.into());
         }
 
+        // Verify that the author doesn't has the proposal in pending list
+        if schema
+            .pending_proposals()
+            .values()
+            .find(|proposal_info| proposal_info.author == author)
+            .is_some()
```

Comment:
As `clippy` suggests, `any` can be used here.

(offtopic: you may want to install [exonum_precheck](https://github.com/popzxc/exonum_precheck) to run all the checks before pushing commits)